plugins/python-build/bin/python-build

```diff
@@ -482,6 +482,13 @@ fetch_git() {
   local git_url="$2"
   local git_ref="$3"
 
+  local fetch_args=(${git_ref})
+  if [ ${#fetch_args[@]} -eq 1 ]; then  # just a branch name
+    fetch_args[0]="+${fetch_args[0]}:${fetch_args[0]}"
+  elif [[ ${#fetch_args[@]} -eq 2 && ${fetch_args[0]} == tag ]]; then
```

Comment:
why does this have to be different from fetching a branch? both branches and tags are just refs to specific commits and you should be able to get either one with the same command.

Reply:
Fetching `+tagname:tagname` doesn't work for tags, which is what python-build does with the refs. git fetch man page documents `tag <tag>` to fetch up to _tag_, and this is what this patch accomplishes.

Granted, it doesn't look at all clean to me and I thought it would've been easier to implement, but this was the smallest change I got working. (Tested with a bunch of micropython fetches and builds.)